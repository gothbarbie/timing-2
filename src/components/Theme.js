import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
