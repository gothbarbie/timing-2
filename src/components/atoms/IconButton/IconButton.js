import React from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const ButtonStyle = styled.button`
  padding: 0;
  margin: 0;
  outline: none;
  background-color: transparent;
  border: none;
  cursor: ${({ onClick }) => (onClick ? 'pointer' : 'none')};
  color: ${({ color, theme }) => color === 'primary' && theme.colors.primary};
  color: ${({ color, theme }) => color === 'white' && theme.colors.white};
  color: ${({ color, theme }) => color === 'gray' && theme.colors.gray};

  &:active,
  &:hover,
  &:focus {
    color: ${({ theme }) => theme.colors.primary};
    color: ${({ theme, hoverColor }) =>
      hoverColor === 'gray' && theme.colors.gray};
  }
`

// NOTE! All icons used must be manually added at src/index.js
const IconButton = ({ color, hoverColor, icon, size = 'lg', onClick }) => (
  <ButtonStyle color={color} hoverColor={hoverColor} onClick={onClick}>
    <FontAwesomeIcon icon={icon} size={size} />
  </ButtonStyle>
)

export default IconButton
