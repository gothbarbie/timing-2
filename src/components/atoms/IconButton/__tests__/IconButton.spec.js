import React from 'react'
import { shallow } from 'enzyme'
import { theme } from '../../../Theme'

import IconButton from '../IconButton'

describe('generic/molecules', () => {
  it('renders', () => {
    const component = shallow(<IconButton theme={theme} />)

    expect(component).toMatchSnapshot()
  })
})
