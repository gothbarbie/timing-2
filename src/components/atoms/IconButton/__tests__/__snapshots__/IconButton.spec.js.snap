// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/molecules renders 1`] = `
<styled.button>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={null}
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="lg"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</styled.button>
`;
